dask/delayed.py

```diff
@@ -208,11 +209,15 @@ def delayed(obj, name=None, pure=False):
     """
     if isinstance(obj, Delayed):
         return obj
+
     task, dasks = to_task_dasks(obj)
-    root = not dasks
-    name = name or '%s-%s' % (type(obj).__name__, tokenize(task, pure=True))
-    dasks.append({name: task})
-    return Delayed(name, dasks, pure=pure, root=root)
+
+    if not dasks and callable(obj):
+        return DelayedFunction(obj, pure=pure, prefix=prefix)
```

Comment:
I'd feel a bit better if this was:

```python
if not dasks:
    return DelayedRoot(obj, pure=pure, prefix=prefix)    # Or some other class name
else:
    ....
```

No harm is done if the object being wrapped isn't callable, and the type of the output of delayed is easier to reason about (imo).